azure/blob-storage/variables.tf

```diff
@@ -0,0 +1,155 @@
+variable "resource_group_name" {
+  type = string
+}
+
+variable "name" {
+  type = string
+}
+
+variable "region" {
+  type = string
+}
+
+variable "project" {
+  type = string
+}
+
+variable "environment" {
+  type = string
+}
+
+variable "lifecycle_rules" {
+  type = map(object({
+    prefix_match                                               = list(string)
+    blob_types                                                 = list(string)
+    tier_to_cool_after_days_since_modification_greater_than    = optional(number, -1)
+    tier_to_archive_after_days_since_modification_greater_than = optional(number, -1)
+    delete_after_days_since_modification_greater_than          = optional(number, -1)
+  }))
+  default = {}
+}
+
+variable "user_assigned_identity_ids" {
+  type    = list(string)
+  default = []
+}
+
+# BlobStorage: for storing unstructured object data, such as text or binary data
+# BlockBlobStorage: premium performance storage account type designed specifically for storing block blobs and append blobs.
+# FileStorage: built for high-performance file shares.
+# Storage: Legacy, support blobs, files, queues, tables, and disks.
+# StorageV2: Latest, support blobs, files, queues, tables, and disks
+variable "account_kind" {
+  type    = string
+  default = "StorageV2"
+
+  validation {
+    condition     = contains(["BlobStorage", "BlockBlobStorage", "FileStorage", "Storage", "StorageV2"], var.account_kind)
+    error_message = "Must be either BlobStorage, BlockBlobStorage, FileStorage, Storage or StorageV2"
+  }
+}
+
+variable "account_tier" {
+  type    = string
+  default = "Standard"
+
+  validation {
+    condition     = contains(["Standard", "Premium"], var.account_tier)
+    error_message = "Must be either Standard or Premium"
+  }
+}
+
+# https://learn.microsoft.com/en-us/azure/storage/common/storage-redundancy
+variable "account_replication_type" {
+  type = string
+  # LRS: Least expensive. copies your data synchronously three times within a single physical location in the primary region
+  # ZRS: copies your data synchronously across three Azure AZs in the primary region.
+  # # Secondary Region
+  # GRS: LRS (Primary Region) + LRS in secondary region.
+  # GZRS: ZRS (Primary Region) + LRS in secondary region
+  default = "LRS"
```

Comment:
`GZRS` costs roughly twice as much as `LRS`, however, if we store e.g. 100 GiB, the cost difference is 4€ vs. 2€, which is minimal.

I thus suggest we go with `GZRS` as default value for higher reliability.